### PR TITLE
Agent ha support

### PR DESCRIPTION
**Pull request by siddontang**

### CODE REVIEW

File: cmd/agent/agent_test.go [hunk 1]
```diff
@@ -281,7 +295,62 @@ func (s *testAgentSuite) testProxy(c *C) {
...
+	time.Sleep(3 * time.Second)
+
+	// first test slave ha
+	procs := s.testGetProcs(c, s.agentStoreSlave)
+	// first stop agent then kill procs
+
+	s.testStopAgent(c, s.agentStoreSlave)
+	s.testKillProcs(c, procs)
+
+	time.Sleep(5 * time.Second)
```
Comment (ngaut) [inline cmd/agent/agent_test.go]: Is 5 seconds too long?

Comment (siddontang) [inline cmd/agent/agent_test.go]: I don't have a good idea to determine a proper wait time for all tests like this, badly.

**Comment by qiuyesuifeng** [comment]
LGTM.
@ngaut  any suggestion? can this be merged?

File: cmd/agent/agent_test.go (hunk 1 above)
Comment (xiang90) [inline cmd/agent/agent_test.go]: @siddontang Do you know exactly what are you waiting for? If you wait for a processes to be killed, you can check its status.

Comment (siddontang) [inline cmd/agent/agent_test.go]: Not simple, e.g, for proxy, we will wait its status online in coordinator, so we should first check the status, then wait some time, and go on.

Comment (xiang90) [inline cmd/agent/agent_test.go]: I usually move this kind of stuff out of unit case to an integration test. Waiting for a long time during test decreases the developing experience a lot, especially for an open source project.

Comment (siddontang) [inline cmd/agent/agent_test.go]: Yes, I know, so I move agent test out of unit test too, separately using `make agent_test` :-)

**Comment by ngaut** [comment]
Let's merge it now.

**Pull request merged**